byway/src/main/webapp/routepage.js

```diff
@@ -12,55 +12,86 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+ 
+// copies of recommendations and route, stored as sets for synchronous updating
+let recs = new Set();
 
-// copies of recommendations and selected stops, stored as arrays for synchronous updating
-let recs = [];
-let stops = []; // TODO: Make set to avoid duplicates
+// holds stops and destinations
+let route = [];
 
+// holds destinations 
+let destinations = []; 
+ 
+// object that communicates with the GMaps API service
+let directionsService;
+ 
+// object that renders display results on the map
+let directionsRenderer;
+ 
+// TODO: get from Trip key
+let start = "";
+let end = "";
+ 
 if (document.readyState === 'loading') {  // Loading hasn't finished yet
   document.addEventListener('DOMContentLoaded', loadData);
+  console.log("loading");
 } else {  // `DOMContentLoaded` has already fired
   loadData();
 }
-
-/** Used to restore stops and recommendations upon load or refresh */
+ 
+/** Used to restore route and recommendations upon load or refresh */
 function loadData(){
+  console.log("loadData");
   getRecsOnload();
-  getStopsOnload();
+  getRouteOnload();
 }
-
+ 
 /** Initializes map on the page */
 function initMap() {
-  const directionsService = new google.maps.DirectionsService();
-  const directionsRenderer = new google.maps.DirectionsRenderer();
-  let start = new google.maps.LatLng(37.7699298, -122.4469157);
-  let end = new google.maps.LatLng(37.7683909618184, -122.51089453697205);
+  directionsService = new google.maps.DirectionsService();
+  directionsRenderer = new google.maps.DirectionsRenderer();
+ 
   let mapOptions = {
     zoom: 14,
-    center: start
+    center: new google.maps.LatLng(40.730610, -73.935242) // coordinates of NYC
   }
   const map = new google.maps.Map(document.getElementById('map'), mapOptions);
   directionsRenderer.setMap(map);
-  document.getElementById("route").addEventListener("click", function() {
-    calcRoute(directionsService, directionsRenderer, start, end);
+  
+}
+
+/** Displays route containing waypoints overtop the map. */
+function calcRoute() {
+  let request = {
+    origin:  start,
+    destination: end,
```

Comment:
Does having these as explicit origin/destination not prevent optimizeWaypoints from moving them into the middle of the route?

Reply:
Yes it does, but I removed start/end from the route list because if they are they will become waypoints. I don't think having generateRoute perform another request is necessary, will remove.